yamllint/rules/quoted_strings.py

```diff
@@ -63,18 +65,34 @@
    ::
 
     foo: 'bar'
+
+#. With ``quoted-strings: {quote-type: single, required: only-when-needed, needed-extra-regex: ^%.*%$ }``
+
+   the following code snippet would **PASS**:
+   ::
+
+    foo: '%bar%'
+
+   the following code snippet would **FAIL**:
+   ::
+
+    foo: 'bar'
```

Comment:
Can you also add:
```
   the following code snippet would **FAIL**:
   ::

    foo: %bar%
```
?

Reply:
Added in dd1db1c12c54a64baa349afed48770ec1f42c6f2.